charts/datadog/templates/_container-agent.yaml

```diff
@@ -148,6 +148,8 @@
     - name: DD_CHECKS_TAG_CARDINALITY
       value: {{ .Values.datadog.checksCardinality | quote }}
     {{- end }}
+    - name: DD_ORCHESTRATOR_EXPLORER_ENABLED
+      value: {{ (include "should-enable-k8s-resource-monitoring" .) | quote }}
```

Comment:
the cluster checks should not run in the "Daemonset" agent pods, but instead in the dedicated clusterchecks deployment, so instead, the envvar should be set here: https://github.com/DataDog/helm-charts/blob/main/charts/datadog/templates/agent-clusterchecks-deployment.yaml#L126